firewood/src/proof.rs

```diff
@@ -173,34 +179,24 @@ impl<N: AsRef<[u8]> + Send> Proof<N> {
                 if does_not_match {
                     return Ok((None, Nibbles::<0>::new(&[]).into_iter()));
                 }
+                let data = n.chd_encoded().ok_or(ProofError::InvalidData)?;
 
-                let sub_proof = if term {
-                    SubProof {
-                        encoded: data,
-                        hash: None,
-                    }
-                } else {
-                    self.generate_subproof(data)?
-                };
+                let sub_proof = self.generate_subproof(data.to_vec())?;
 
                 Ok((sub_proof.into(), key_nibbles))
             }
-
-            BRANCH_NODE_SIZE if key_nibbles.size_hint().0 == 0 => Err(ProofError::NoSuchNode),
-
-            BRANCH_NODE_SIZE => {
+            NodeType::Branch(n) => {
+                if key_nibbles.size_hint().0 == 0 {
+                    return Err(ProofError::NoSuchNode);
+                }
```

Comment:
Nit:
Every time you have to add more indentation to your code, you're adding complexity for a reader/review. I think it's much better to keep your methods/functions as flat as possible.
```suggestion

            NodeType::Branch(_) if key_nibbles.size_hint().0 == 0  => return Err(ProofError::NoSuchNode),
            
            NodeType::Branch(n) => {
```